Core/src/org/sleuthkit/autopsy/datasourceprocessors/xry/XRYCallsFileParser.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.datasourceprocessors.xry;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.List;
+import java.util.logging.Level;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.Content;
+import org.sleuthkit.datamodel.TskCoreException;
+
+/**
+ * Parses XRY Calls files and creates artifacts.
+ */
+final class XRYCallsFileParser extends AbstractSingleKeyValueParser {
+
+    private static final Logger logger = Logger.getLogger(XRYCallsFileParser.class.getName());
+
+    //Human readable name of this parser.
+    private static final String PARSER_NAME = "XRY Calls";
```

Comment:
Did Brian specify that each type of file parser should associates its name with the attributes it parses? If not, I think the precedents would indicate we use the name of the XRY DSP here - it is the "module."

Reply:
No, I wasn't thinking much about this. I'll change these